### ISSUE TITLE
record_group_name in logging_presets.py: Forgot to account for hash_map?

### ISSUE THREAD

**Issue by HSinger04** [KevinMusgrave/pytorch-metric-learning#591]
I believe I have found a bug in [def record_group_name](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/src/pytorch_metric_learning/utils/logging_presets.py#L360) i.e. that `record_group_name` does not account for that its return value might have to be altered through [RecordKeeper's hash_map](https://github.com/KevinMusgrave/record-keeper/blob/master/src/record_keeper/record_keeper.py#L21). 

The following concrete bug occured for me:   
During my training, PML tries to create a SQL table (I hope I am using the right terminology, I am not familiar with SQL) which is originally supposed to be called `'accuracies_normalized_trunk_WithAMPGlobalEmbeddingSpaceTester_level_0_VAL_vs_SAMPLES'`. However, since it gets hashed by [c_f.hash_if_too_long](https://github.com/KevinMusgrave/record-keeper/blob/master/src/record_keeper/record_keeper.py#L42), it gets hashed to `'accuracies_788b7d1303988de301ae61d98b1eed07e8d8eede2084e972f8c739b32170c1a4_SAMPLES'`, which gets used as the actual table name. Later on, PML tries to access the table (e.g. in https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/src/pytorch_metric_learning/utils/logging_presets.py#L286), but tries to do so with the original value `'accuracies_normalized_trunk_WithAMPGlobalEmbeddingSpaceTester_level_0_VAL_vs_SAMPLES'` and thus can't find the table, leading to bugs like returning `[], None` in `get_accuracies_of_best_epoch` even though there are recorded past accuracies.

For now, I have fixed the issue for myself by changing `def record_group_name` as follows:
```
def record_group_name(self, tester, split_name):
    base_record_group_name = self.base_record_group_name(tester)
    query_set_label = split_name.upper()
    reference_sets_label = "_and_".join(
        [name.upper() for name in sorted(tester.reference_split_names[split_name])]
    )
    if reference_sets_label == query_set_label:
        reference_sets_label = "self"
    record_group_name = f"{base_record_group_name}_{query_set_label}_vs_{reference_sets_label}"
    # NEW STUFF - BEGIN
    if record_group_name in self.record_keeper.hash_map.values():
        for key, val in self.record_keeper.hash_map.items():
            if val == record_group_name:
                record_group_name = key
    # NEW STUFF - END
    return record_group_name
```

**Comment by KevinMusgrave** [comment]
Thanks for catching this bug. Your code looks reasonable. Do you mind opening a pull request? You just need to fork this repo, checkout the dev branch, make your change, go to [pull requests](https://github.com/KevinMusgrave/pytorch-metric-learning/pulls) and make a new pull request to merge the change.

On a related note, maybe it would make sense to have a flag to allow turning off the hash map. I added the hash map because in one of my other projects, some of the recorded data is in very deeply nested objects, resulting in very long record group names. So converting to a hashed string made sense. But it would be nice to be able to turn it off. What do you think?

**Comment by HSinger04** [comment]
Thx for the precise instructions, I have created a pull request now :) 

Don't see any downside to it and would be less confusing for the users, so sounds like a good idea.

**Comment by KevinMusgrave** [comment]
Fixed in v2.0.1
